### ISSUE TITLE
[BUG]multicore mode does not work

### ISSUE THREAD

**Issue by seechange0710** [htseq/htseq#85]
**Software versions**
Specify what versions of the following you are using:
- `HTSeq` so I use HTSeq-count with version 2.0.7
- `Python` 3.12.1
- operating system macOS Sonoma 14.4.1
- downstream of HISAT2 aligner

**Describe the bug**
Hi dear HTseq team,

I kept receiving an error message as I was trying to run `htseq-count` using `multicore (-n)` mode on multiple bam files (>1).
The error message as blow:
<img width="743" alt="Screenshot 2024-06-12 at 19 42 46" src="https://github.com/htseq/htseq/assets/84936409/3d42c993-f8df-4213-8825-e33951f65211">

The code I used as below:
>htseq-count -m union -c ~/Documents/Brassica_napus/NN_data/3.Alignment/Results/Bn3CL-count-sorted.csv -f bam -s no -t exon -i gene_id --additional-attr gene_name -r pos -n 6 ~/Documents/Brassica_napus/NN_data/3.Alignment/Results/*_sortedP.bam ~/Documents/Brassica_napus/NN_data/3.Alignment/Preparation/Gene_annotation/Brassica_napus.AST_PRJEB5043_v1.59.gtf.gz

before I post this report here, I also checked the codes of the version where `-n --nprocesses` was **for the first time added** and found, however, in test file of the **current version** the corresponding block was commented out and marked with "multicore makes a mess on travis". 
>
    # Testing multiple cores on travis makes a mess
    #{'call': [
    #    'htseq-count',
    #    '-n', '2',
    #    f'{data_folder}/bamfile_no_qualities.sam',
    #    f'{data_folder}/bamfile_no_qualities.gtf',
    #    ],
    # 'expected_fn': f'{data_folder}/bamfile_no_qualities.tsv'},
I cannot figure out if the error arose because I didn't put the arguments in the right order in the command that I used (e.g. wildcard does not work properly or files should be provided in a list with comma delimiter etc.) or simply because the `multicore` function is at the moment not available?

Many thanks in advance!

**Comment by iosonofabio** [comment]
Thanks for reporting. I'll take a look but an not terribly familiar with multiprocess.

Could you make a PR? That would speed up things significantly

**Comment by iosonofabio** [comment]
the commented out test is skipped because, AFAIK, travis and now github actions are not that happy about multicore testing at their expense.

I've tested that command locally with twice the same sam file:

```
htseq-count -n 2 example_data/bamfile_no_qualities.sam example_data/bamfile_no_qualities.sam example_data/bamfile_no_qualities.gtf
```

and did not get the error you mention. Also, `freeze_support` should be for windows only, not sure why it would show up on a mac.

Do you have a linux machine that you can test this on?

**Comment by seechange0710** [comment]
> the commented out test is skipped because, AFAIK, travis and now github actions are not that happy about multicore testing at their expense.
> 
> I've tested that command locally with twice the same sam file:
> 
> ```
> htseq-count -n 2 example_data/bamfile_no_qualities.sam example_data/bamfile_no_qualities.sam example_data/bamfile_no_qualities.gtf
> ```
> 
> and did not get the error you mention. Also, `freeze_support` should be for windows only, not sure why it would show up on a mac.
> 
> Do you have a linux machine that you can test this on?

Hi, thanks for the reply.

I just tested the same code above on my Mac and it generated the same error which seems like to be a common issue maybe for Mac operation system.
I'll test my code on a linux computer and come back soon.

**Comment by iosonofabio** [comment]
I believe this was closed by #86
